src/main/java/com/feed_the_beast/ftbu/ranks/Rank.java

```diff
@@ -142,8 +142,9 @@ public void fromJson(JsonElement e)
             {
                 String id = a.get(i).getAsString();
                 char firstChar = id.charAt(0);
-                boolean b = firstChar == '-';
-                permissions.put((firstChar == '-' || firstChar == '+') ? id.substring(1) : id, b);
+                boolean not_allowed = firstChar == '-';
```

Comment:
This looks like a logical error and also with this it works.
Think about it — if first symbol is `-`, then you should put this permission with `false` in the map, because above you make a direct check with this map.